pkg/tfsandbox/tofu.go

```diff
@@ -65,11 +67,53 @@ func NewTofu(ctx context.Context, workdir Workdir) (*Tofu, error) {
 		return nil, fmt.Errorf("error creating a tofu executor: %w", err)
 	}
 
+	cacheDir, err := getPluginCacheDir()
+	if err != nil {
+		return nil, fmt.Errorf("error getting plugin cache dir: %w", err)
+	}
+	// Use a common cache directory for provider plugins
+	env := envMap(os.Environ())
+	env["TF_PLUGIN_CACHE_DIR"] = cacheDir
+	if err := tf.SetEnv(tfexec.CleanEnv(env)); err != nil {
```

Comment:
What does CleanEnv do, curious?

Reply:
`terraform-exec` has certain environment variables that it does not allow you to set this way. It make you set them via a specific method e.g. `tf.SetLog("debug")` sets the `TF_LOG` env var. If you try to set it in `tf.SetEnv` it will throw an error. Their `CleanEnv` function removes these type of variables.